system/include/simd128.h

```diff
@@ -0,0 +1,729 @@
+/*
+WebAssembly SIMD128 Intrinsics
+*/
+
+#include <stdint.h>
+
+// User-facing type
+typedef int32_t v128_t __attribute__((vector_size(16), __aligned__(16)));
+
+// Internal types determined by clang builtin definitions
+typedef int32_t __v128_u __attribute__((__vector_size__(16), __aligned__(1)));
+typedef char  __i8x16 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef unsigned char __u8x16 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef short __i16x8 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef unsigned short __u16x8 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef int __i32x4 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef unsigned int __u32x4 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef long long __i64x2 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef unsigned long long __u64x2 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef float __f32x4 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef double __f64x2 __attribute__((__vector_size__(16), __aligned__(16)));
+
+#define __DEFAULT_FN_ATTRS __attribute__((__always_inline__, __nodebug__, __target__("simd128"), __min_vector_width__(128)))
+
+// v128 wasm_v128_load(void* mem)
+static __inline__ v128_t __DEFAULT_FN_ATTRS wasm_v128_load(const void* __mem) {
+  struct __wasm_v128_load_struct {
+    __v128_u __v;
+  } __attribute__((__packed__, __may_alias__));
```

Comment:
why is a `struct` needed here? you could just cast the pointer to `__v128_u*`

Reply:
This is due to a rather long discussion above about providing unaligned accesses while avoiding undefined behavior. I ended up copying how xmmintin.h does its loads and stores, which involves this helper struct.

Reply:
Maybe adding a comment here pointing to the `xmmintin.h` implementation would help.

IIRC, the previous implementation did not have a `__v128_u` type that's `__aligned(1)__`. With this type, `return *((__v128_u*)__mem);` should just work.